### DISCUSSION TITLE
Where do we get the user input in examples/main.cpp?

### DISCUSSION THREAD

**Discussion by buyukakyuz** [ggml-org/llama.cpp#2462] [Q&A]

**Comment by klosax** [comment]
Here:

https://github.com/ggerganov/llama.cpp/blob/11f3ca06b8c66b0427aab0a472479da22553b472/examples/main/main.cpp#L742
https://github.com/ggerganov/llama.cpp/blob/11f3ca06b8c66b0427aab0a472479da22553b472/examples/common.cpp#L953
